stock_location_orderpoint_cleanup/models/stock_location_orderpoint.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2024 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+import logging
+
+from odoo import _, api, models
+
+from odoo.addons.queue_job.job import identity_exact
+
+_logger = logging.getLogger(__name__)
+
+
+class StockLocationOrderpoint(models.Model):
+
+    _inherit = "stock.location.orderpoint"
+
+    def _get_moves_to_cleanup_domain(self) -> list:
+        self.ensure_one()
+        return [
+            ("location_orderpoint_id", "=", self.id),
+            ("state", "not in", ("done", "cancel")),
+            ("quantity_done", "<=", 0),
+            ("picking_id.printed", "=", False),
+        ]
+
+    def _get_moves_to_cleanup(self):
+        self.ensure_one()
+        moves = self.env["stock.move"].search(self._get_moves_to_cleanup_domain())
+        return moves
```

Comment:
I would not restrict this method to one orderpoint. This is not needed. 

```suggestion
        return [
            ("location_orderpoint_id", "in", self.ids),
            ("state", "not in", ("done", "cancel")),
            ("quantity_done", "<=", 0),
            ("picking_id.printed", "=", False),
        ]

    def _get_moves_to_cleanup(self):
        moves = self.env["stock.move"].search(self._get_moves_to_cleanup_domain())
        return moves
```